unit/test-suffix-si-parse.c

```diff
@@ -11,51 +13,61 @@
 
 static int test_rc;
 
-static void check_num(const char *val, int lbas, __u64 exp, __u64 num)
+static void check_num(const char *val, __u64 exp, __u64 num)
 {
 	if (exp == num)
 		return;
 
-	printf("ERROR: printing {%s} (lbas %d), got '%llu', expected '%llu'\n",
-	       val, lbas, (unsigned long long)num, (unsigned long long)exp);
+	printf("ERROR: printing {%s},  got '%llu', expected '%llu'\n",
+	       val, (unsigned long long)num, (unsigned long long)exp);
 
 	test_rc = 1;
 }
 
 struct tonum_test {
 	const char *val;
-	int lbas;
-	const __u64 exp;
+	const uint64_t exp;
+	int ret;
 };
 
 static struct tonum_test tonum_tests[] = {
-	{ "11995709440", 512, 11995709440 },
-	{ "1199570940", 512, 1199570940 },
-	{ "6.14T", 512, 11992187500 },
-	{ "6.14T", 520, 11807692307 },
-	{ "6.14T", 4096, 1499023437 },
-	{ "6.14", 512, 0 },
-	{ "6.14#", 512, 0 },
+	{ "11995709440", 11995709440, 0 },
+	{ "1199570940", 1199570940, 0},
+	{ "234.567M", 234567000, 0 },
+	{ "1.2k", 1200, 0 },
+	{ "6.14T", 6140000000000, 0 },
+	{ "123.4567k", 123456, 0 },
+	{ "12345.6789101112M", 12345678910, 0},
+	{ "6.14", 6, 0 },
+	{ "6.14#", 0, -EINVAL },
+	{ "2,33", 0, -EINVAL },
+	{ "3..3", 0, -EINVAL },
 };
 
 void tonum_test(struct tonum_test *test)
 {
-	__u64 num;
-	bool suffixed;
-
-	num = suffix_si_parse(test->val, &suffixed);
-
-	if (suffixed)
-		num /= test->lbas;
+	char *endptr;
+	uint64_t num;
+	int ret;
+
+	ret = suffix_si_parse(test->val, &endptr, &num);
+	if (ret != test->ret) {
+		printf("ERROR: converting {%s} failed\n", test->val);
+		test_rc = 1;
+		return;
+	}
+	if (ret)
+		return;
 
-	check_num(test->val, test->lbas, test->exp, num);
+	check_num(test->val, test->exp, num);
 }
 
 int main(void)
 {
 	unsigned int i;
 
 	test_rc = 0;
+	setlocale(LC_NUMERIC, "C");
```

Comment:
About the locale is it enough to test "C"? It seems so but to make sure let me confirm it.

Reply:
Should be okay, I've checked the documentation. Obviously, we could add a test which tests for ',' decimal point but I guess but there are better things to test :)